commit ad7a4c803fb072ac76ef49a74876bf72df04f391
Author: Sage Weil <sweil@redhat.com>
Date:   Tue Jun 13 15:08:19 2017 -0400

    mon,mgr: pass new-style health checks from mgr's PGMap
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index 7b9086c4800..096ab5bcb52 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -992,15 +992,14 @@ void DaemonServer::send_report()
 	}
       }
 
-      // FIXME: reporting health detail here might be a bad idea?
       cluster_state.with_osdmap([&](const OSDMap& osdmap) {
 	  // FIXME: no easy way to get mon features here.  this will do for
 	  // now, though, as long as we don't make a backward-incompat change.
 	  pg_map.encode_digest(osdmap, m->get_data(), CEPH_FEATURES_ALL);
 	  dout(10) << pg_map << dendl;
-	  pg_map.get_health(g_ceph_context, osdmap,
-			    m->health_summary,
-			    &m->health_detail);
+
+	  pg_map.get_health_checks(g_ceph_context, osdmap,
+				   &m->health_checks);
 	});
     });
   // TODO? We currently do not notify the PyModules
